### PR TITLE
feat: [CDS-72961]: DOC for terraform apply step with encrypt terrafor…

### PR DESCRIPTION
**Pull request by vlica-harness**
…m json outputs

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Comment by KimberlyFields** [comment]
@roshnisarangadharan since Michael is on PTO, can you review this one?

**Review by roshnisarangadharan — APPROVED** [review]
LGTM. Let me know if its good to merge, @vlica-harness.

File: docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-apply-step.md [hunk 1]
```diff
@@ -463,6 +463,26 @@ TF_VAR_alist='[1,2,3]'
 ```
 You can use Harness encrypted text for values. See [Add Text Secrets](/docs/platform/Secrets/add-use-text-secrets).
 
+## Encrypt the Terraform Apply JSON outputs
+
+Currently, this feature is behind the flag, `CDS_ENCRYPT_TERRAFORM_APPLY_JSON_OUTPUT`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-apply-step.md:468]:
```suggestion
Currently, this feature is behind the feature flag, `CDS_ENCRYPT_TERRAFORM_APPLY_JSON_OUTPUT`. Contact [Harness Support](mailto:support@harness.io) to enable this feature.
```

File: docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-apply-step.md [hunk 2]
```diff
@@ -463,6 +463,26 @@ TF_VAR_alist='[1,2,3]'
...
+
+```
+{
+"test-output-name1": {
+"sensitive": false,
+"type": "string",
+"value": "test-output-value1"
+}
+```
+This secret expression can be used in other steps that can parse, validate,and extract the required output values from the JSON.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-apply-step.md:484]:
```suggestion
This secret expression can be used in other steps that can parse, validate, and extract the required output values from the JSON.
```

File: docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-apply-step.md [hunk 3]
```diff
@@ -463,6 +463,26 @@ TF_VAR_alist='[1,2,3]'
 ```
 You can use Harness encrypted text for values. See [Add Text Secrets](/docs/platform/Secrets/add-use-text-secrets).
 
+## Encrypt the Terraform Apply JSON outputs
+
+Currently, this feature is behind the flag, `CDS_ENCRYPT_TERRAFORM_APPLY_JSON_OUTPUT`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+
+In Optional Configuration it's required to provide a secret manager for the "Encrypt json output" field.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-apply-step.md:470]:
```suggestion
In **Optional Configuration** of the Terraform Apply step, you must provide a secret manager for the **Encrypt json output** field.
```

File: docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-apply-step.md [hunk 4]
```diff
@@ -463,6 +463,26 @@ TF_VAR_alist='[1,2,3]'
 ```
 You can use Harness encrypted text for values. See [Add Text Secrets](/docs/platform/Secrets/add-use-text-secrets).
 
+## Encrypt the Terraform Apply JSON outputs
+
+Currently, this feature is behind the feature flag, `CDS_ENCRYPT_TERRAFORM_APPLY_JSON_OUTPUT`. Contact [Harness Support](mailto:support@harness.io) to enable this feature.
+
+In **Optional Configuration** of the Terraform Apply step, you must provide a secret manager for the **Encrypt json output** field.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-apply-step.md:470]:
```suggestion
In the Terraform Apply step's **Optional Configuration**, you must select a secret manager for the **Encrypt json output** field to encrypt and provide the Terraform JSON output as a secret expression.
```

**Review by michaelcretzman — APPROVED** [review]

**Pull request merged**